### PR TITLE
Refactor JSONSchema and URL checks

### PR DESCRIPTION
**Pull request by mondeja**
- Deduplicates URL regex checks and definitions.
- Uses `const` instead of single value enum for custom license types in JSONSchema.
- Includes *.jsonschema.json* file in JS package.
- Improves types for `getIconsData*` functions of SDK.

### CODE REVIEW

File: .jsonschema.json [hunk 1]
```diff
@@ -617,7 +616,7 @@
     },
     "duplicate": {
       "$id": "#duplicate",
-      "description": "A \"dup\" brand",
+      "description": "A duplicated brand",
```
Comment (PeterShaggyNoble) [inline .jsonschema.json:619]:
```suggestion
      "description": "A brand that uses the same icon",
```

**Comment by mondeja** [comment]
The changes proposed on https://github.com/simple-icons/simple-icons/pull/10573#discussion_r1511863468 must be backported to this PR before the merge so we remove the URL regex entirely.

**Comment by mondeja** [comment]
The previous comment (now hidden) has no sense because in this PR I'm using the same regex for JSONSchema check and in the code. The regex can't be removed.

**Comment by adamrusted** [comment]
Hey @mondeja - are you okay to resolve the conflicts on this ready for next week? We can look to merge in as part of the major release then :)

**Comment by LitoMore** [comment]
As I remember we are planning to move the SDK part to a module.

**Comment by mondeja** [comment]
Don't worry, I'll resolve the conflicts before the Wednesday regardless of whether we move the SDK to a separate npm package. In that case, the changes proposed by this PR must be included. Feel free to open the PR to separate the SDK meanwhile @LitoMore. If not, I'll open it before the weekend if I have time for it, anyways we can always postpone that change for v13.

**Comment by LitoMore** [comment]
Ok, I will try to create the new repository for SDK.

**Comment by mondeja** [comment]
> Ok, I will try to create the new repository for SDK.

As you want but we can release it from this same repository.

**Comment by LitoMore** [comment]
> As you want but we can release it from this same repository. Probably this approach is better because there are some files in simple-icons/simple-icons that the SDK needs to work.

Okay, we need to change this repository to [workspaces](https://docs.npmjs.com/cli/v10/using-npm/workspaces). The folder structure needs some changes.

**Comment by mondeja** [comment]
This is ready @LitoMore, would you mind to review it?

File: sdk.d.ts [hunk 2]
```diff
@@ -63,9 +63,9 @@ export type IconData = {
 /* The next code is autogenerated from sdk.mjs */
 /* eslint-disable */
 
-export const URL_REGEX: RegExp;
 export const SVG_PATH_REGEX: RegExp;
 export function getDirnameFromImportMeta(importMetaUrl: string): string;
+export function urlRegex(jsonschemaPath?: string): Promise<RegExp>;
```
Comment (LitoMore) [inline sdk.d.ts:68]: We used a `fs.readFile` for the JSON schema file. This means the `.jsonschema.json` file will be missed during the tree-shaking.

This means that our `urlRegex()` cannot be used in some scenarios.

Comment (mondeja) [reply sdk.d.ts:68]: I've added `!.jsonschema.json` to `.npmignore`, so will be included in the package. This is the same behaviour of other functions like `getIconsData`, `getThisPartyExtensions` or `getThirdPartyLibraries`, which read from the icons data file and the README.

**Review by LitoMore — APPROVED** [review]

**Pull request merged**